app-frontend/src/app/components/map/annotateSidebarItem/annotateSidebarItem.controller.js

```diff
@@ -0,0 +1,117 @@
+/* globals L, $*/
+
+export default class AnnotateSidebarItemController {
+    constructor(
+        $log, $scope, $location, $anchorScroll, $timeout,
+        mapService
+    ) {
+        'ngInject';
+        this.$log = $log;
+        this.$scope = $scope;
+        this.$location = $location;
+        this.$anchorScroll = $anchorScroll;
+        this.$timeout = $timeout;
+
+        this.getMap = () => mapService.getMap(this.mapId);
+    }
+
+    $onInit() {
+        this.isEditing = false;
+    }
+
+    $onChanges(changeObj) {
+        if (changeObj.scrollId && changeObj.scrollId.currentValue) {
+            this.scrollToItem(changeObj.scrollId.currentValue);
+        }
+    }
+
+    scrollToItem(id) {
+        let newHash = 'anchor' + id;
+        if (this.$location.hash() !== newHash) {
+            this.$location.hash(newHash);
+        } else {
+            this.$anchorScroll();
+        }
+    }
+
+    onChangeItemAppearance(annotation) {
+        this.onChangeAnnotationStroke({
+            'annotation': annotation
+        });
+    }
+
+    onAnnotationClone(annotation) {
+        this.onCloneAnnotation({
+            'geometry': annotation.geometry,
+            'label': annotation.properties.label,
+            'description': annotation.properties.description
+        });
+    }
+
+    /* eslint-disable no-underscore-dangle */
+    onAnnotationEdit(annotation) {
+        this.isEditing = true;
+
+        this.scrollToItem(annotation.properties.id);
+
+        this.onUpdateAnnotationStart({'annotation': annotation});
+    }
+    /* eslint-enable no-underscore-dangle */
+
+    onAnnotationDelete(annotation) {
+        this.onDeleteAnnotation({
+            'id': annotation.properties.id,
+            'label': annotation.properties.label
+        });
+    }
+
+    addAnnotation(annotation) {
+        if (annotation.properties.label !== 'Label name...') {
+            $('#_value').css('border-color', '#fff');
```

Comment:
Angular has built in validation, we don't need to manually set css like this.
see https://www.w3schools.com/angular/angular_validation.asp and this example: https://www.w3schools.com/angular/tryit.asp?filename=try_ng_validate_show (you can use ng-class instead of ng-show for the border styles)